src/librustc/middle/traits/select.rs

```diff
@@ -1555,62 +1512,41 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                 // "opened" unsized/existential type (one that has
                 // been dereferenced)
                 match bound {
-                    ty::BoundCopy |
-                    ty::BoundSync |
-                    ty::BoundSend => {
-                        Ok(If(vec!(ty)))
-                    }
+                    ty::BoundCopy | ty::BoundSync | ty::BoundSend => Ok(If(vec!(ty))),
 
-                    ty::BoundSized => {
-                        Err(Unimplemented)
-                    }
+                    ty::BoundSized => Err(Unimplemented),
                 }
             }
-            ty::ty_err => {
-                Ok(If(Vec::new()))
-            }
+            ty::ty_err => Ok(If(Vec::new())),
 
-            ty::ty_infer(ty::FreshTy(_)) |
-            ty::ty_infer(ty::FreshIntTy(_)) => {
+            ty::ty_infer(ty::FreshTy(_))
+            | ty::ty_infer(ty::FreshIntTy(_)) =>
                 self.tcx().sess.bug(
                     &format!(
                         "asked to assemble builtin bounds of unexpected type: {}",
-                        self_ty.repr(self.tcx()))[]);
-            }
+                        self_ty.repr(self.tcx()))[]),
         };
 
         fn nominal<'cx, 'tcx>(this: &mut SelectionContext<'cx, 'tcx>,
                               bound: ty::BuiltinBound,
                               def_id: ast::DefId,
                               types: Vec<Ty<'tcx>>)
-                              -> Result<BuiltinBoundConditions<'tcx>,SelectionError<'tcx>>
+                              -> Result<BuiltinBoundConditions<'tcx>, SelectionError<'tcx>>
         {
             // First check for markers and other nonsense.
             let tcx = this.tcx();
             match bound {
-                ty::BoundSend => {
-                    if Some(def_id) == tcx.lang_items.managed_bound() {
-                        return Err(Unimplemented)
-                    }
-                }
+                ty::BoundSend if Some(def_id) == tcx.lang_items.managed_bound()
+                    => Err(Unimplemented),
 
-                ty::BoundCopy => {
-                    return Ok(ParameterBuiltin)
-                }
+                ty::BoundCopy => Ok(ParameterBuiltin),
 
-                ty::BoundSync => {
-                    if
-                        Some(def_id) == tcx.lang_items.managed_bound() ||
-                        Some(def_id) == tcx.lang_items.unsafe_cell_type()
-                    {
-                        return Err(Unimplemented)
-                    }
-                }
+                ty::BoundSync if Some(def_id) == tcx.lang_items.managed_bound() ||
+                                 Some(def_id) == tcx.lang_items.unsafe_cell_type()
+                    => Err(Unimplemented),
 
-                ty::BoundSized => { }
+                _ => Ok(If(types)),
```

Comment:
this could allow for future code to be missed, should i create a manual fallback with all possible cases instead?